test/functional/cmdLineTests/gcsuballoctests/gcsuballoctests.xml

```diff
@@ -24,10 +24,11 @@
 
 <!DOCTYPE suite SYSTEM "cmdlinetester.dtd">
 
-<suite id="J9 GC  Tests" timeout="60">
+<suite id="J9 GC Suballocator Tests" timeout="60">
  
- <!-- Following tests are for compressedrefs platforms. 
-	Because -Xgc:suballocatorInitialSize= and -Xgc:suballocatorCommitSize= options are defined only for compressedrefs. -->
+ <!-- Following tests are for compressedrefs platforms because -Xgc:suballocatorInitialSize=,
+      -Xgc:suballocatorCommitSize=, -Xgc:suballocatorIncrementSize=, -Xgc:suballocatorQuickAllocDisable
+      and -Xgc:suballocatorQuickAllocEnable options are defined only for compressedrefs. -->
```

Comment:
I don't see anything in the code that limits the options to compressedrefs mode: Please explain.

Reply:
By "the code" do you mean in places like [this](https://github.com/eclipse/omr/blob/576481c1ac4e34da29886a444d19574bc029156e/gc/base/GCExtensionsBase.hpp#L747) not being behind `#if defined(OMR_GC_COMPRESSED_POINTERS)` guards?

The `suballocator...` options are only ever used by code guarded by `#if defined(OMR_GC_COMPRESSED_POINTERS)`. I assumed the previous `suballoctor...` command-line options were still defined for non-compressed refs builds for Java command invocation compatibility between non-compressed refs builds and compressed refs builds, so I followed that decision.

Reply:
The options are recognized unconditionally (there's no `#if defined(OMR_GC_COMPRESSED_POINTERS)` guarding that code in `mmparseXgc.cpp`), but I see the use of `suballocatorIncrementSize` in eclipse/omr#7414 is guarded - perhaps the field declaration should be as well?

Reply:
Yes, that was the point I was trying to make in my previous comment.

I think it is better if the options aren't guarded " for Java command invocation compatibility between non-compressed refs builds and compressed refs builds" like I mentioned.

Is your opinion that we guard them? And also @dmitripivkine what is yours?

Reply:
I wasn't suggesting that the option parsing be conditional, just the presence of the fields.

Reply:
Right. So then in the `playlist.xml` file for this test, do you know how I could specify that it should only be run on compressed reference builds?

Reply:
I would prefer not to wrap field as well as option recognition in this case. The motivation is we prefer to recognize the option always to allow customers to use command line flexibly. And guarding inside option recognition does not make much sense, so my vote is not use wrapping for field declaration too.

Reply:
If we agree with Dmitri, the most recent push addresses all of your comments @keithc-ca

Reply:
It's not my preference to have fields that are not used. The more serious problem is using when they should not be, but I can live with the fields being present even if they're not used.

Reply:
> how I could specify that it should only be run on compressed reference builds?

Perhaps @llxia can help with that.

Reply:
That is only necessary if we `#if defined(OMR_GC_COMPRESSED_POINTERS)` guard the parsing. Otherwise, we can just leave it as it is.

Reply:
No. The possibility is that someone adds uses of the fields thinking they're relevant when they're not. If the field were conditional, those uses would have to be conditional as well. We should let the compiler help us with things like that, but as I said, I won't insist on making the fields conditional (which would be an OMR committer's decision).

Reply:
I was referring to changing the tests to run only for compressed refs. I agree with you on the point you just made.

Reply:
Looking at the tests again, I think you're right - they seem mostly about option validation which happens independently of whether compressed refs are in use or not.